language-support/java/codegen/BUILD.bazel

```diff
@@ -90,22 +92,41 @@ daml_lf_target_versions = ["1.0", "1.1", "1.3"]
 ####  Integration Tests
 ########################################################
 [
-    daml_to_java(
-        name = "integration-tests-model-%s" % target,
-        main_src = "src/it/daml/Lib.daml",
-        srcs = glob(["src/it/daml/**/*.daml"]),
-        target = target,
-    )
+    #
+    # Compile the DAR files with and without a package to allow
+    # DAML-LF version agnostic tests to be run using generated
+    # Java classes with the same qualified name, while the DAML-LF
+    # specific tests run using Java classes generated with a
+    # package name qualified by the DAML-LF version.
+    #
+    [
+        daml_compile(
+            name = "integration-tests-model-%s" % target,
+            main_src = "src/it/daml/Lib.daml",
+            srcs = glob(["src/it/daml/**/*.daml"]),
+            target = target,
+        ),
+        dar_to_java(
+            name = "integration-tests-model-noprefix-%s" % target,
+            src = "integration-tests-model-%s.dar" % target,
```

Comment:
Could you change it so that this actually uses the target notation with a leading `:`? Ideally I'd like to make a proper macro out of it that verifies that it is an actual label. That way it's clear on the use-site that this points to a label and not just a file that "happens to be at the right location".